docker/jvm/Dockerfile

```diff
@@ -0,0 +1,78 @@
+###############################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+
+ARG GOLANG_VERSION=1.21.1
+
+FROM golang:${GOLANG_VERSION} AS build-ub
+WORKDIR /build
+RUN useradd --no-log-init --create-home --shell /bin/bash appuser
+COPY --chown=appuser:appuser resources/ub/ ./
+RUN go build -ldflags="-w -s" ./ub.go
+USER appuser
+RUN go test ./...
+
+
+FROM eclipse-temurin:17-jre
```

Comment:
I suggest we go with 21 from the start.

Reply:
We were waiting for [this](https://hub.docker.com/layers/library/eclipse-temurin/21-jre/images/sha256-0e3f813fbdcd0b62e58c3ad984da29d96e0b78cede4578d2d1d0f8391f1f136c?context=explore) to release. Will be updating it post testing